app/react/Users/components/UserSidePanel.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Icon } from 'UI';
+import { Translate } from 'app/I18N';
+import { ConfirmButton, SidePanel } from 'app/Layout';
+import { UserRole } from 'shared/types/userSchema';
+import { UserSchema } from 'shared/types/userType';
+import MultiSelect from 'app/Forms/components/MultiSelect';
+import { UserGroupSchema } from 'shared/types/userGroupType';
+import { PermissionsList } from 'app/Users/components/PermissionsList';
+
+export interface UserSidePanelProps {
+  user: UserSchema;
+  users: UserSchema[];
+  groups: UserGroupSchema[];
+  opened: boolean;
+  closePanel: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onSave: (user: UserSchema) => void;
+  onDelete: (user: UserSchema) => void;
+  onReset2fa: (user: UserSchema) => void;
+  onResetPassword: (user: UserSchema) => void;
+}
+
+export const UserSidePanel = ({
+  user,
+  users,
+  groups,
+  opened,
+  closePanel,
+  onSave,
+  onDelete,
+  onReset2fa,
+  onResetPassword,
+}: UserSidePanelProps) => {
+  const [userToSave, setUserToSave] = useState(user);
+  const [permissionsModalOpened, setPermissionsModalOpened] = useState(false);
+  const [selectedGroups, setSelectedGroups] = useState(
+    user.groups ? user.groups.map(group => group._id!.toString()) : []
+  );
+
+  const { register, handleSubmit, errors } = useForm();
+
+  const availableGroups = groups;
+
+  const handleInputChange = (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
+    const { name, value } = event.currentTarget;
+    const updatedUser = { ...userToSave, [name]: value };
+    setUserToSave(updatedUser);
+  };
+
+  const onChangeHandler = (groupIds: string[]) => {
+    setSelectedGroups(groupIds);
+    const updatedGroups = groups
+      .filter(group => groupIds.includes(group._id as string))
+      .map(group => ({
+        _id: group._id!,
+        name: group.name,
+      }));
+    setUserToSave({ ...userToSave, groups: updatedGroups });
+  };
+
+  const isDuplicated = (nameVal: string) =>
```

Comment:
Technically this should be called `isUnique`, right?
Or in other case remove the `!` at the begining (or double negation `!!`)
We might be wrong :) but please take a look!

Reply:
You're right, `isUnique` is the word that describes correctly the function.